scripts/build_llvm_project.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+
+import argparse
+import multiprocessing
+import os
+from pathlib import Path
+import platform
+import settings
+import shutil
+import subprocess
+import sys
+
+def default_toolchain():
+  system = platform.uname()[0]
+  machine = platform.uname()[4]
+  toolchain = os.path.join(settings.THIS_DIR,
+                           'cmake',
+                           f'toolchain_{system.lower()}_{machine.lower()}.cmake')
+  return toolchain if os.path.isfile(toolchain) else None
+
+def get_arguments():
+  parser = argparse.ArgumentParser(
+    description="Help configure and build classic-flang-llvm-project")
+
+  buildopt = parser.add_argument_group('general build options')
+  buildopt.add_argument('-t', '--target', metavar='ARCH', choices=['X86', 'AArch64', 'PowerPC'], default='X86',
+                        help='Control which targets are enabled (%(choices)s) (default: %(default)s)')
+  buildopt.add_argument('-p', '--install-prefix', metavar='PATH', nargs='?', default=os.path.join(settings.LLVM_DIR,'..','install'), const=False,
+                        help='install directory (default: %(default)s) ')
+  buildopt.add_argument('-j', '--jobs', metavar='N', type=int, default=os.cpu_count(),
+                        help='number of parallel build jobs (default: %(default)s)')
+  buildopt.add_argument('--toolchain', metavar='FILE', default=default_toolchain(),
+                        help='specify toolchain file (default: %(default)s)')
+  buildopt.add_argument('--builddir', metavar='DIR', default='build',
+                        help=f'specify build directory (default: {settings.LLVM_DIR}/%(default)s)')
+  buildopt.add_argument('--clean', action='store_true', default=False,
+                        help='clean build')
+  buildopt.add_argument('--build-type', metavar='TYPE', default='Release',

```

Comment:
I think the options should support a single-letter short form, similar to the ones in `build-llvm-project.sh`. `--build-type` and `--use-ccache` are especially important since a user will probably be typing them quite frequently.